### ISSUE TITLE
String phaser

### ISSUE THREAD

**Issue by jpcima** [jpcima/string-machine#11]
Since the string synthesizer is so commonly paired with the EHX small stone phaser, it will be interesting if this embeds its own analog phaser effect.
More so if the phaser is made compatible with stereo, to connect in series with the chorus.

This code implements a vintage phaser. [Phase100.dsp.txt](https://github.com/jpcima/string-machine/files/3474572/Phase100.dsp.txt)

```
Kiiski, R., Esqueda, F., & Välimäki, V. (2016).
Time-variant gray-box modeling of a phaser pedal.
In 19th International Conference on Digital Audio Effects (DAFx-16).
```

It's a 10-stage and quite different from the desired phaser; but the reference paper gives a method to extract modulation from any phaser.
It will serve if a volunteer owner of small stone or similar wants to contribute some measurements.

**Comment by SpotlightKid** [comment]
Maybe ask @brummer10 whether he has already modelled a small stone or a similar phaser for guitarix?

**Comment by jpcima** [comment]
Yes it's a good idea. I'm going to share a few resources regarding phasers.

Small stone is a 4-stage allpass filter with feedback loop.
Filters have cutoff modulated with an identical LFO.
CA3094 "OTA" are used for varying the filters.

- Structure is similar to MXR phase 90 analyzed here : https://www.electrosmash.com/mxr-phase90
- Other article on phase 90.
```
Eichas, F., Fink, M., Holters, M., & Zölzer, U. (2014, September).
Physical Modeling of the MXR Phase 90 Guitar Effect Pedal. In DAFx (pp. 153-158).
```
- Small stone schematic
![moddedSmlStnSchm](https://user-images.githubusercontent.com/17614485/62736553-e9c7f080-ba2d-11e9-9377-db7f9ea5cb7f.gif)

**Comment by jpcima** [comment]
other thing I thought worth noting: supposed interesting behavior of the phaser pedal running under low power condition.
![Capture du 2019-08-09 09-07-55](https://user-images.githubusercontent.com/17614485/62760824-41457b00-ba85-11e9-829d-c889e8f9bb7e.png)

**Comment by jpcima** [comment]
Here is "Ross phaser" a similar one, except using LM13600 OTA, which is a more common electronic part.
These OTA have spice models, unlike the CA3094, so maybe an electronic simulation would be possible from it.
http://www.tonepad.com/project.asp?id=25
